### DISCUSSION TITLE
关于报错后，游戏停止运行的问题

### DISCUSSION THREAD

**Discussion by constly** [ejoy/ant#92] [General]
目前只要有lua报错，游戏就会立刻停止运行。实际上有些报错影响不大，游戏正式上线后，也不太可能发现有lua报错就宕机吧。

这个问题怎么优化比较好呢，我想到的一个折中办法是在调用system实现的stage时，加一个xpcall，如果有报错只是立刻终止本次stage的调用... 

这个问题大家怎么看呢？

**Comment by cloudwu** [comment]
这个感觉需要实践找到合适的方法。或许还要根据游戏具体做。

像 factorio 的插件是出错就弹框屏蔽掉的。
